src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -5146,13 +5152,16 @@ impl Coordinator {
         session: &Session,
         optimizer_notices: &Vec<OptimizerNotice>,
     ) {
-        if self
-            .catalog
-            .system_config()
-            .enable_notices_for_index_too_wide_for_literal_constraints()
-        {
-            let humanizer = self.catalog().for_session(session);
-            for optimizer_notice in optimizer_notices {
+        let humanizer = self.catalog().for_session(session);
+        for optimizer_notice in optimizer_notices {
+            let system_vars = self.catalog.system_config();
+            let notice_enabled = match optimizer_notice {
+                OptimizerNotice::IndexTooWideForLiteralConstraints(..) => {
+                    system_vars.enable_notices_for_index_too_wide_for_literal_constraints()
```

Comment:
@ggnall / @uce Do we need a special feature flag for each notice type? From a code maintenance point of view I would prefer to have one.

Reply:
I'm thinking that it might be useful to have separate feature flags, because it can very well happen that we decide that the `IndexTooWideForLiteralConstraints` notice would be a bit too spammy, but the `IndexKeyEmpty` is ok.

Reply:
> Yes, but in two months we have 10 features and 10 feature flags that live in the code forever and are never removed this flexibility also has its downsides.

Having only one feature flag for all the notices would be problematic, because there is a high chance that we'll arrive at different decisions for different notices regarding whether we'll want to show it as a notice.

> It's probably fine as a switch when rolling out features individual notices, but I wonder if there is a plan to remove them after some time.

Hopefully, we'll arrive at some permanent decisions after a while regarding which notices we want to show, and then we can remove the feature flags.

> You can also have a single enable_notices configuration parameter (synced from LaunchDarkly) of type JSON where you would provide a list of wildards of the notice types that you want to allow or deny in general or for a specific environment.

My feeling is that aggregating them to a JSON field could mean more maintenance cost than having separate feature flags, but I'm not sure. How much is the maintenance cost of a feature flag? A feature flag feels quite simple to me.

Reply:
> How much is the maintenance cost of a feature flag?

Its approximately:

The engineering effort is repeated per notice:
1. Code:
    - Adding the lint to `feature_flags!`
    - Guarding the notice output with the new `system_vars` condition in the `sequence_` methods.
2. Adding the corresponding flag to LD.
3. Removing the code changes from (1).
4. Removing the code changes from (4).

The operational effort is:
1. Go to LD and tweak the value of a single boolean flag.

With one feature flag the engineering effort is going to be paid once upfront for all notice types.

1. Code
    - Create a `optimizer_notice` system variable of type `[String]` and code that decides whether a specific notice type is enabled or disabled based on the `optimizer_notice` parameter (interpret each `String` as an allow or deny rule).
2. Add the corresponding flag to LD (using type "JSON" or "String").
 

However, the operational effort _might be_ higher:
1. For each new combination of configurations, we will need to create a new `optimizer_notice` variant.

Reply:
Since I have not used LD before, I'd prefer to keep the LD side simple at the cost of a few extra lines of code. The coding effort is really minimal: writing the code and potentially deleting it later shouldn't take more than 15 mins per notice.

Reply:
(It's 6-8 lines of code per flag, depending on formatting.)

Reply:
OK.

Reply:
Late, but I would go with this approach. Flags should be deprecated as part of the feature lifecycle.
>It's probably fine as a switch when rolling out features individual notices, but I wonder if there is a plan to remove them after some time.

+1
>I'm thinking that it might be useful to have separate feature flags, because it can very well happen that we decide that the IndexTooWideForLiteralConstraints notice would be a bit too spammy, but the IndexKeyEmpty is ok.